test/Qowaiv.Validation.DataAnnotations.UnitTests/DataAnnotationsValidatorTest.cs

```diff
@@ -113,12 +113,36 @@ public void Validate_NestedModelWithInvalidChildren_with_error()
                 Id = Guid.NewGuid(),
                 Children = new[]
                 {
-                    new NestedModelWithChildren.ChildModel{ Name = "Valid Name" },
-                    new NestedModelWithChildren.ChildModel()
+                    new NestedModelWithChildren.ChildModel{ ChildName = "Valid Name" },
+                    new NestedModelWithChildren.ChildModel(),
                 }
             };
             DataAnnotationsAssert.WithErrors(model,
-                ValidationMessage.Error("The Name field is required.", "Children[1].Name"));
+                ValidationMessage.Error("The ChildName field is required.", "Children[1].ChildName"));
+        }
+
+        [Test]
+        public void Validate_NestedModelWithInvalidGrandchildren_with_error()
```

Comment:
I would also include a test for children without indexes.
e.g. "Item1[0].Item2.Item3"
and "Item1.Item2[0].Item3"
and "Item1.Item2.Item3"